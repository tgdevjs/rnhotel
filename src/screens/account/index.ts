export * from './Account';
export * from './AccountGuest';
export * from './AccountMenu';
export * from './JoinModal';
export * from './SignInModal';
