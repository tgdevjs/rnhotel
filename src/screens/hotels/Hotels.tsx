import React, { Component } from "react";
import { ImageBackground, FlatList, StyleSheet, View } from "react-native";
import { NavigationScreenProp } from "react-navigation";
import Icons from "react-native-vector-icons/MaterialCommunityIcons";
import LinearGradient from "react-native-linear-gradient";
import { TouchableHighlight } from "react-native-gesture-handler";

import { HotelTitle } from "../../components";

type Props = {
  navigation: NavigationScreenProp<any, any>;
};

const styles = StyleSheet.create({
  iconRow: {
    flexDirection: "row"
  },
  item: {
    justifyContent: "flex-end",
    height: 160,
    width: "100%"
  },
  itemTitleBar: {
    marginHorizontal: 8,
    marginVertical: 8
  },
  icons: {
    margin: 2
  },
  itemSeparator: {
    backgroundColor: "white",
    height: 10
  }
});

const data = Array.from({ length: 30 }, (v, i) => ({
  key: String(i),
  name: `hotel-${i}`,
  imageSrc: "http://lorempixel.com/320/280/city",
  price: "$120"
}));

const renderIconRow = () => {
  const icons = [
    "key-variant",
    "bus-side",
    "car",
    "wifi",
    "smoking-off",
    "silverware-fork-knife"
  ].map(v => (
    <Icons
      style={styles.icons}
      key={`${v}`}
      name={v}
      size={25}
      fontWeight="800"
      color="white"
    />
  ));

  return <View style={styles.iconRow}>{icons}</View>;
};

export const Hotels = ({ navigation: { navigate } }: Props) => {
  return (
    <FlatList
      data={data}
      ItemSeparatorComponent={() => <View style={styles.itemSeparator} />}
      renderItem={({ item, item: { name, imageSrc } }) => (
        <TouchableHighlight
          onPress={() => navigate("ConfirmReservation", { hotel: item })}
        >
          <ImageBackground source={{ uri: imageSrc }} style={styles.item}>
            {/* <View style={styles.item}> */}
            <LinearGradient colors={["#192f6a00", "#3b599888", "#4c669fff"]}>
              <View style={styles.itemTitleBar}>
                <HotelTitle title={name} />
                {renderIconRow()}
              </View>
            </LinearGradient>
            {/* </View> */}
          </ImageBackground>
        </TouchableHighlight>
      )}
    />
  );
};
