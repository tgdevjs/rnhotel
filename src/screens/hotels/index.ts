export * from './Hotels';
export * from './ConfirmReservation';
