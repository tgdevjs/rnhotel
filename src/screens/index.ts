export * from './account';
export * from './hotels';
export * from './search';
export * from './stays';
