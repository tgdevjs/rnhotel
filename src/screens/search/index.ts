export * from './Search';
export * from './SearchCalendar';
