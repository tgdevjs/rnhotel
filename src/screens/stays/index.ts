export * from './Stays';
export * from './StayItem';
export * from './StaysList';
export * from './StaysListGuest';
