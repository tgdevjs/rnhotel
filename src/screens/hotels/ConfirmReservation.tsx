import React from 'react';
import { Button, StyleSheet, Text, View } from 'react-native';
import PropTypes from 'prop-types';
import { StackActions, NavigationActions } from 'react-navigation';
import { compose, graphql } from 'react-apollo';
import LinearGradient from 'react-native-linear-gradient';
import cuid from 'cuid';

import { DateDisplay, HotelTitle, SignInButton } from '../../components';
import { withSearchMutation, withSearchQuery } from '../../apollo/client-state/search';
import { withUserQuery } from '../../apollo/client-state/user';
import { reservationsQuery, createReservationMutation } from '../../apollo/queries';
import { getReservationsQueryVariables } from '../../utils';

const styles = StyleSheet.create({
  container: {
    justifyContent: 'space-between',
    flex: 1,
  },
  header: {
    alignItems: 'center',
    justifyContent: 'center',
    height: 200,
  },
  dateDisplay: {
    paddingBottom: 30,
  },
  hotelTitle: {
    color: 'black',
  },
  main: {
    flex: 1,
    justifyContent: 'center',
  },
  buttonSection: {
    marginBottom: 40,
    justifyContent: 'center',
    alignItems: 'center',
  },
  actionSection: {
    margin: 20,
    alignItems: 'center',
  },
  actionText: {
    fontSize: 16,
  },
  labelText: {
    fontSize: 16,
  },
  valueText: {
    fontSize: 20,
    fontWeight: '600',
  },
});

export class ConfirmReservation extends React.Component {
  state = { hasCreatedReservation: false };

  id = cuid();

  renderContent() {
    const {
      createReservation,
      endDay,
      navigation: { dispatch, getParam, navigate, popToTop },
      startDay,
      user: { name },
    } = this.props;
    const { name: hotelName } = getParam('hotel');

    const variables = {
      data: {
        id: this.id,
        name,
        hotelName,
        arrivalDate: startDay,
        departureDate: endDay,
      },
    };
    const { hasCreatedReservation } = this.state;

    // User not logged in
    if (!name) {
      return (
        <View style={styles.buttonSection}>
          <View style={styles.actionSection}>
            <Text style={styles.labelText}>Sign in to continue.</Text>
          </View>
          <SignInButton onPress={() => navigate('SignIn')} />
        </View>
      );
    }

    if (!hasCreatedReservation) {
      return (
        <View style={styles.buttonSection}>
          <View style={styles.actionSection}>
            <Text style={styles.labelText}>User Name:</Text>
            <Text style={styles.valueText}>{name}</Text>
          </View>
          <Button
            title="Book Now"
            onPress={async () => {
              await createReservation({
                variables,
                refetchQueries: [
                  { query: reservationsQuery, variables: getReservationsQueryVariables({ name }) },
                ],
              });
              this.setState({ hasCreatedReservation: true });
            }}
          />
        </View>
      );
    }

    return (
      <View style={styles.buttonSection}>
        <View style={styles.actionSection}>
          <Text style={styles.valueText}>Success!</Text>
          <Text style={styles.labelText}>Your reservation was created.</Text>
        </View>
        <Button title="Start New Search" onPress={() => popToTop()} />
        <Button
          title="View Reservations"
          onPress={() => {
            // Reset Main stack
            dispatch(
              StackActions.reset({
                index: 0,
                actions: [NavigationActions.navigate({ routeName: 'Main' })],
              })
            );
            navigate('Stays');
          }}
        />
      </View>
    );
  }

  render() {
    const {
      endDay,
      navigation: { getParam },
      startDay,
    } = this.props;
    const { name: hotelName } = getParam('hotel');

    return (
      <View style={styles.container}>
        <LinearGradient colors={['#192f6a00', '#3b599888']} style={styles.header}>
          <DateDisplay style={styles.dateDisplay} startDay={startDay} endDay={endDay} />
          <HotelTitle style={styles.hotelTitle} title={hotelName} />
          <Text>{this.id}</Text>
        </LinearGradient>
        <View style={styles.main}>{this.renderContent()}</View>
      </View>
    );
  }
}

export const ConfirmReservationQuery = compose(
  withSearchQuery,
  withSearchMutation,
  withUserQuery,
  graphql(createReservationMutation, { name: 'createReservation' })
)(ConfirmReservation);

ConfirmReservation.propTypes = {
  createReservation: PropTypes.func.isRequired,
  endDay: PropTypes.string.isRequired,
  navigation: PropTypes.shape({}).isRequired,
  startDay: PropTypes.string.isRequired,
  user: PropTypes.shape({
    name: PropTypes.string,
  }).isRequired,
};
