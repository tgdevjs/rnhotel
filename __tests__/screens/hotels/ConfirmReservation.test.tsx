import React from "react";
import { ConfirmReservation } from "../../../src/screens/hotels";

function confirmReservation(props = {}) {
  const defaultProps = {
    navigation: { getParam: jest.fn(() => ({ name: "hotelName" })) },
    user: { name: "userName" },
    ...props
  };
  return shallow(<ConfirmReservation {...defaultProps} />);
}

describe("ConfirmReservation", () => {
  it("renders default", () => {
    expect(confirmReservation()).toMatchSnapshot();
  });
});
