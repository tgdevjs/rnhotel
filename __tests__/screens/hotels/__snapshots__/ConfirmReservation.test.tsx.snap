// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmReservation renders default 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <LinearGradient
    colors={
      Array [
        "#192f6a00",
        "#3b599888",
      ]
    }
    end={
      Object {
        "x": 0.5,
        "y": 1,
      }
    }
    start={
      Object {
        "x": 0.5,
        "y": 0,
      }
    }
    style={
      Object {
        "alignItems": "center",
        "height": 200,
        "justifyContent": "center",
      }
    }
  >
    <DateRange
      dateTextStyle={
        Object {
          "paddingBottom": 30,
        }
      }
    />
    <HotelTitle
      style={
        Object {
          "color": "black",
        }
      }
      title="hotelName"
    />
    <Text />
  </LinearGradient>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginBottom": 40,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "margin": 20,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 16,
            }
          }
        >
          User Name:
        </Text>
        <Text
          style={
            Object {
              "fontSize": 20,
              "fontWeight": "600",
            }
          }
        >
          userName
        </Text>
      </View>
      <Button
        onPress={[Function]}
        title="Book Now"
      />
    </View>
  </View>
</View>
`;
